docs/Escape Hatch.md

```diff
@@ -0,0 +1,309 @@
+# Escape Hatch design
```

Comment:
Given that we call this functionality `escape hatch` in this doc and elsewhere, should we rename the plugin directory to `escape_hatch` rather than `env_escape`?

Reply:
I thought "environment escape" was a better term for it but I can rename to escape_hatch or change the docs to say "Environment escape" everywhere (or "environment escape hatch"); I would prefer one of those two options. Would that work for you?